aspects/schema.go

```diff
@@ -138,7 +138,42 @@ func (s *StorageSchema) Validate(raw []byte) error {
 	return s.topLevel.Validate(raw)
 }
 
+// parseAlternatives tries to parse raw as a JSON list of alternatives types.
+// If the input is not a JSON list, a *json.UnmarshalTypeError is returned.
+func (s *StorageSchema) parseAlternatives(raw json.RawMessage) (*alternativesSchema, error) {
+	var alternatives []json.RawMessage
+	if err := json.Unmarshal(raw, &alternatives); err != nil {
+		return nil, err
+	}
+
+	var alt alternativesSchema
```

Comment:
You could pre-allocate here:
```suggestion
	alt := alternativesSchema{schemas: make([]Schema, 0, len(alternatives))}
```